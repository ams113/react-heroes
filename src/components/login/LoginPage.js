import React, { useContext } from 'react'
import { AuthContext } from '../../auth/authContext'
import { types } from '../../types/types'
import './style.css'

export const LoginPage = ({ history }) => {

    const { dispatch } = useContext( AuthContext )

    const handleLogin = () => {
        // history.push('/');
        // history.replace('/');

        const lastPath = localStorage.getItem('lastPath') || '/';

        dispatch({
            type: types.login,
            payload: {
                name: 'Alfred'
            }
        });

        history.replace( lastPath );
    }

    return (
        <div className="login-container text-center">
            <div className="form-signin">
                <img 
                    className="rounded mb-4" 
                    src={ window.location.origin + '/assets/logo-login.jpg' }  
                    alt="logo" 
                    width="150" 
                    height="150"
                />
                <h1 className="h3 mb-3 font-weight-normal">Heroes Login</h1>
                    
                <input 
                    type="email"
                    id="inputEmail" 
                    className="form-control" 
                    placeholder="Email address" 
                    required 
                />
                
                <input 
                    type="password" 
                    id="inputPassword" 
                    className="form-control" 
                    placeholder="Password" 
                    required 
                />
                <div className="checkbox mb-3">
                    <label>
                        <input 
                            type="checkbox" 
                            value="remember-me" 
                        /> 
                                Remember me
                    </label>
                </div>
                    <button 
                        className="btn btn-lg btn-dark btn-block" 
                        type="submit"
                        onClick={ handleLogin }
                    >
                            Sign in
                        </button>
                    <p className="red mt-5 mb-3">&copy; 2020-2021</p>
            </div>
        </div>
    )
}
