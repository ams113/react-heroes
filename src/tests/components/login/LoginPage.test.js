import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import { AuthContext } from '../../../auth/authContext';
import { LoginPage } from '../../../components/login/LoginPage';
import { types } from '../../../types/types';
import { contextLogout } from '../../fixtures/contextLogout';


describe('Testing Components: LoginPage', () => {

    const history = {
        replace: jest.fn()
    };

    const wrapper = mount(
        <AuthContext.Provider value={ contextLogout }>
            <MemoryRouter>
                <LoginPage history={ history }/>
            </MemoryRouter>
        </AuthContext.Provider>
    );

    test('should render correctly', () => {

        expect( wrapper ).toMatchSnapshot();
    });

    test('should perform dispatch and navigation ', () => {
        
        wrapper.find('button').simulate('click');

        expect( contextLogout.dispatch ).toHaveBeenCalledWith({
            type: types.login,
            payload: {
                name: 'Alfred'
            }
        })
        expect( history.replace ).toHaveBeenCalledWith('/');

    });

    test('should get lastPath from the localStorege', () => {
        
        const handleClick = wrapper.find('button').prop('onClick');
        localStorage.setItem('lastPath', '/dc');
        handleClick();
        expect( history.replace ).toHaveBeenCalledWith('/dc');
    });
    
});
