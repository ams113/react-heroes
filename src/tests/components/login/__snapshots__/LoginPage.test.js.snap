// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Components: LoginPage should render correctly 1`] = `
<div
  className="login-container text-center"
>
  <div
    className="form-signin"
  >
    <img
      alt="logo"
      className="rounded mb-4"
      height="150"
      src="http://localhost/public/assets/logo-login.jpg"
      width="150"
    />
    <h1
      className="h3 mb-3 font-weight-normal"
    >
      Heroes Login
    </h1>
    <input
      className="form-control"
      id="inputEmail"
      placeholder="Email address"
      required={true}
      type="email"
    />
    <input
      className="form-control"
      id="inputPassword"
      placeholder="Password"
      required={true}
      type="password"
    />
    <div
      className="checkbox mb-3"
    >
      <label>
        <input
          type="checkbox"
          value="remember-me"
        />
        Remember me
      </label>
    </div>
    <button
      className="btn btn-lg btn-dark btn-block"
      onClick={[Function]}
      type="submit"
    >
      Sign in
    </button>
    <p
      className="red mt-5 mb-3"
    >
      Â© 2020-2021
    </p>
  </div>
</div>
`;
